src/commands/lambda/flare.ts

```diff
@@ -383,16 +436,28 @@ export const createDirectories = (
  * Gets the LOG_STREAM_COUNT latest log stream names, sorted by last event time
  * @param cwlClient CloudWatch Logs client
  * @param logGroupName name of the log group
+ * @param startMillis start time in milliseconds or undefined if no start time is specified
+ * @param endMillis end time in milliseconds or undefined if no end time is specified
  * @returns an array of the last LOG_STREAM_COUNT log stream names or an empty array if no log streams are found
  * @throws Error if the log streams cannot be retrieved
  */
-export const getLogStreamNames = async (cwlClient: CloudWatchLogsClient, logGroupName: string) => {
-  const command = new DescribeLogStreamsCommand({
+export const getLogStreamNames = async (
+  cwlClient: CloudWatchLogsClient,
+  logGroupName: string,
+  startMillis: number | undefined,
+  endMillis: number | undefined
+) => {
+  const config = {
     logGroupName,
-    limit: LOG_STREAM_COUNT,
     descending: true,
     orderBy: OrderBy.LastEventTime,
-  })
+    limit: DEFAULT_LOG_STREAMS,
+  }
+  const rangeSpecified = startMillis !== undefined && endMillis !== undefined
```

Comment:
This cannot be simplified to `!startMillis && !endMillis` because if `startMillis = 0`, then `rangeSpecified` will incorrectly have the value of `false`.